### PR TITLE
Add pokes system with .POK file format (Feature 3b)

### PR DESCRIPTION
**Pull request by ikari-pl**
## Summary
- New `poke` IPC commands for loading and applying WinAPE .POK cheat files
- Parses standard .POK format with game groups, multi-value pokes, and undo support
- Supports direct memory writes for ad-hoc poking
- New files: `src/pokes.h`, `src/pokes.cpp`

## IPC Commands
- `poke load <path>` — load a .POK file
- `poke list` — list loaded games and pokes with applied status
- `poke apply <game> <poke|all>` — apply poke(s) to memory
- `poke unapply <game> <poke>` — restore original values
- `poke write <addr> <value>` — direct memory write

## Test plan
- [x] 20 new unit tests (parser, apply/unapply, edge cases)
- [x] All 260 tests pass (240 existing + 20 new)
- [x] `make clean && make` — clean build

### CODE REVIEW

File: src/pokes.cpp [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    if (games_.empty()) return "no games found in file";
+    return ""; // success
+}
+
+int PokeManager::apply(size_t game_idx, size_t poke_idx, WriteFn write_fn, ReadFn read_fn) {
+    if (game_idx >= games_.size()) return -1;
+    auto& game = games_[game_idx];
+    if (poke_idx >= game.pokes.size()) return -1;
+    auto& poke = game.pokes[poke_idx];
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/pokes.cpp:94]: Fixed — added early return if poke.applied is already true, plus a new unit test verifying double-apply doesn't corrupt original_value.

File: src/koncepcja_ipc_server.cpp [hunk 2]
```diff
@@ -1613,6 +1614,102 @@ std::string handle_command(const std::string& line) {
...
+        for (size_t pi = 0; pi < games[gi].pokes.size(); pi++) {
+          resp += "  Poke: " + games[gi].pokes[pi].description;
+          resp += " [" + std::to_string(games[gi].pokes[pi].values.size()) + " value";
+          if (games[gi].pokes[pi].values.size() != 1) resp += "s";
+          resp += "]";
+          if (games[gi].pokes[pi].applied) resp += " [applied]";
+          resp += "\n";
+        }
+      }
+      return resp;
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/koncepcja_ipc_server.cpp]: Fixed — using std::ostringstream now.

File: src/koncepcja_ipc_server.cpp [hunk 3]
```diff
@@ -1613,6 +1614,102 @@ std::string handle_command(const std::string& line) {
...
+    }
+    if (parts[1] == "apply" && parts.size() >= 3) {
+      size_t game_idx;
+      try {
+        game_idx = static_cast<size_t>(std::stoul(parts[2]));
+      } catch (...) {
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/koncepcja_ipc_server.cpp]: Fixed — replaced catch(...) with specific std::invalid_argument and std::out_of_range catches.

File: src/pokes.cpp [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+int PokeManager::apply_all(size_t game_idx, WriteFn write_fn, ReadFn read_fn, int* total_values) {
+    if (game_idx >= games_.size()) return -1;
+    auto& game = games_[game_idx];
+    int pokes_applied = 0;
+    int values_total = 0;
+    for (size_t i = 0; i < game.pokes.size(); i++) {
+        int n = apply(game_idx, i, write_fn, read_fn);
+        if (n >= 0) {
```
[the remark this answers is not in the document]
Comment (ikari-pl) [reply src/pokes.cpp]: Fixed — changed to `if (n > 0)` to match the updated apply() behavior.
